astropy/table/table.py

```diff
@@ -2918,8 +2918,12 @@ def sort(self, keys=None, reverse=False):
             prev_frozen = sort_index._frozen
             sort_index._frozen = True
 
-        for col in self.columns.values():
-            col[:] = col.take(indexes, axis=0)
+        for name, col in self.columns.items():
+            new_col = col.take(indexes, axis=0)
+            try:
+                col[:] = new_col
+            except Exception:
```

Comment:
could you make it to the specific exception? 
```suggestion
            except TypeError:
```

Reply:
This exception can be raised by arbitrary user-defined mixin columns, so I have no idea what exception class they might raise in this case.